### PR TITLE
feat(iot): Add missing moduleContent field to ConfigurationContent class

### PR DESCRIPTION
**Pull request by timtay-microsoft**
#2185

### CODE REVIEW

File: iothub/service/src/Configurations/ConfigurationContent.cs [hunk 1]
```diff
@@ -12,16 +12,28 @@ namespace Microsoft.Azure.Devices
...
         public IDictionary<string, IDictionary<string, object>> ModulesContent { get; set; }
 
 #pragma warning restore CA2227 // Collection properties should be read only
 
         /// <summary>
-        /// Gets or sets Module Configurations
+        /// Gets or sets the configurations to be applied on device modules
+        /// </summary>
+        [JsonProperty(PropertyName = "moduleContent")]
+#pragma warning disable CA2227 // Collection properties should be read only
```
Comment (azabbasi) [inline iothub/service/src/Configurations/ConfigurationContent.cs:30]: Do these have any benefits over the attribute version of warning suppression? As far as I know using the preprocessor directives and the using the suppression attribute are alternatives. Attributes read and look much cleaner specially for property warning suppressions.

File: iothub/service/src/Configurations/ConfigurationContent.cs [hunk 2]
```diff
@@ -12,16 +12,28 @@ namespace Microsoft.Azure.Devices
...
+        public IDictionary<string, object> ModuleContent { get; set; }
```
Comment (azabbasi) [inline iothub/service/src/Configurations/ConfigurationContent.cs:31]: Interesting, so we never even had it? do we know if this was added at some point and we just never got the message or we originally missed it? Not that it really matters here, I am just curious.

**Review by azabbasi — APPROVED** [review]
LGTM

File: iothub/service/src/Configurations/ConfigurationContent.cs (hunk 2 above)
Comment (timtay-microsoft) [reply iothub/service/src/Configurations/ConfigurationContent.cs:31]: I'm not sure. I imagine the SDK team wasn't given a full spec for this feature, and this field was lost in the shuffle somehow

**Pull request merged**